src/components/NewCommentForm.tsx

```diff
@@ -47,25 +46,31 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
-      name: !name,
-      email: !email,
-      body: !body,
-    });
+    if (selectedPost) {
```

Comment:
Issue: The check for `selectedPost` being truthy is correct, but it might be more robust to ensure `selectedPost.id` is also defined, as it is used later in the code.